driver-core/src/main/com/mongodb/internal/authentication/AwsCredentialHelper.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.authentication;
+
+import com.mongodb.AwsCredential;
+import com.mongodb.MongoInternalException;
+import com.mongodb.lang.NonNull;
+import org.bson.BsonDocument;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+public final class AwsCredentialHelper {
+
+    public static AwsCredential obtainFromEnvironment() {
+        if (System.getenv("AWS_ACCESS_KEY_ID") != null) {
+            return obtainFromEnvironmentVariables();
+        } else {
+            return obtainFromEc2OrEcsResponse();
+        }
+    }
+
+    private static AwsCredential obtainFromEnvironmentVariables() {
```

Comment:
This makes it clear that all the credential pieces come from variables, or none do.